docs/modules/ROOT/pages/reference/quality-functional.adoc

```diff
@@ -0,0 +1,74 @@
+= Functional
+
+== Provisioning of a Service Happens Fully Automated
+
+Source::
+User
+
+Stimulus::
+Orders a service via the provided APIs
+
+Environment::
+AppCatalog enabled Kubernetes cluster
+
+Artifact::
+AppCatalog Service
+
+Response::
+Service created
+
+Response measure::
+Within a few minutes, see service for specific definition.
```

Comment:
We can do that, but 30 minutes feel like a bit long. Maybe we also have to differentiate between single-instances and cluster? Single-instances will probably be within 5 minutes, and clusters could take up to 15 minutes.

Reply:
Thinking about it again, I'm not sure it even makes sense to put a deadline on provisioning. Considering we now only go for "best-effort" and SLA-backed support options, there's no reason that the service has to be running after x minutes for the best-effort case. And in case of guaranteed availability, I guess this should somehow be alert-based.

Reply:
I'm not sure if an alert would make sense here. It could lead to some false positives, probably if a customer orders something while cluster maintenance is ongoing.

Also, how would that be actionable? If the provisioning takes longer due to infrastructure or longer image pull times, then this is rather a symptom  of another issue and not an issue itself.

Reply:
Service provisioning time isn't part of the SLA, so there is nothing we have to guarantee. But IMHO tracking the time how long it takes to provision a service should be important for us. I'd say let's go for 30 minutes and learn how all the services behave when we have them. We can then revisit this number.

Reply:
From a Software architecture textbook PoV, it's the quality requirements that make up the software architecture. In this case however, I fail to see the impact this time deadline has on the overall architecture. What happens if the provisioning fails in x minutes and how do we ensure they stay under the limit? For me service provisioning is something a là "it takes as long as it needs". In case of abnormal conditions this requirement hardly covers such cases anyway.
If we just track the time for dashboard purposes, I don't think we need to formulate a quality requirement here at all since for me, these are very strong and formal. If we just go for the dashboard purpose, a simple functional requirement (feature) like "internal dashboard for service provisioning time" somewhere else would make more sense.

Reply:
For me the time it takes until a service is provisioned is a quality requirement. If it takes ages, I might choose a different service provider. We should take appropriate measures to have a service provisioned in due time, and we should make sure we meet this time constraint, otherwise the quality requirement is violated, and the service needs to be investigated.

Reply:
I see. Then I would suggest something like this:
```suggestion
Response::
Service ready

Response measure::
Service ready depending on service architecture in normal conditions:
- Standalone: within 10 minutes
- Replicated: within 20 minutes
- Clustered: within 30 minutes
- Cloud instance: within 20 minutes
```